engine/src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -491,6 +490,35 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
     runTransaction(action)
   }
 
+  override def upsertExecutionInfo(workflowId: WorkflowId,
+                                   callKey: BackendCallKey,
+                                   keyValues: Map[String, Option[String]])
+                                  (implicit ec: ExecutionContext): Future[Unit] = {
+    import ExecutionIndex._
+
+    val action = for {
+      executionResult <- dataAccess.executionsByWorkflowExecutionUuidAndCallFqnAndShardIndexAndAttempt(
+        workflowId.toString, callKey.scope.fullyQualifiedName, callKey.index.fromIndex, callKey.attempt).result.head
+
+      _ <- DBIO.sequence(keyValues map upsertExecutionInfo(executionResult.executionId.get))
+    } yield ()
+
+    runTransaction(action)
+  }
+
+  private def upsertExecutionInfo(executionId: Int)(keyValue: (String, Option[String]))
+                                 (implicit ec: ExecutionContext): DBIO[Unit] = {
+    val (key, value) = keyValue
+    for {
+      backendUpdate <- dataAccess.executionInfoValueByExecutionAndKey(executionId, key).update(value)
+      _ <- backendUpdate match {
+        case 0 => dataAccess.executionInfosAutoInc += ExecutionInfo(executionId, key, value)
+        case 1 => DBIO.successful(Unit)
+        case _ => DBIO.failed(new RuntimeException(s"Unexpected backend update count $backendUpdate"))
```

Comment:
Very cool.  I hope to make `EXECUTION_INFO` append-only [one day soon](#596), but this is still cool.  :smile: